CMakeLists does not check for presence of protobuf library in default mode, but requires it for compilation

CMake passes in default mode even when protobuf is not installed. But later during compilation I see errors about missing headers.


`cmake ..`:

```
-- The C compiler identification is GNU 9.2.0
-- The CXX compiler identification is GNU 9.2.0
-- Check for working C compiler: /bin/cc
-- Check for working C compiler: /bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /bin/c++
-- Check for working CXX compiler: /bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found OpenSSL: /usr/lib/libcrypto.so (found version "2.0.0")
-- Could NOT find libprotobuf-c (missing: NATS_PROTOBUF_INCLUDE_DIR NATS_PROTOBUF_LIBRARY)
-- Configuring done
-- Generating done
-- Build files have been written to: /home/rk/Repositories/nats.c/buils
```

`make`:

```
Scanning dependencies of target nats
[  1%] Building C object src/CMakeFiles/nats.dir/asynccb.c.o
In file included from /home/rk/Repositories/nats.c/src/stan/stanp.h:19,
                 from /home/rk/Repositories/nats.c/src/stan/conn.h:17,
                 from /home/rk/Repositories/nats.c/src/asynccb.c:18:
/home/rk/Repositories/nats.c/src/stan/protocol.pb-c.h:7:10: fatal error: protobuf-c/protobuf-c.h: No such file or directory
    7 | #include <protobuf-c/protobuf-c.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [src/CMakeFiles/nats.dir/build.make:63: src/CMakeFiles/nats.dir/asynccb.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1013: src/CMakeFiles/nats.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
```

Then I have configured it without streaming (`cmake .. -DNATS_BUILD_STREAMING=OFF
`) and it builds without errors. It's fine for me since I do not seed streaming.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -85,6 +85,8 @@ if(NATS_BUILD_STREAMING)
     SET(NATS_PROTOBUF_DIR "" CACHE STRING
       "An optional hint to a directory for finding `libprotobuf-c`"
     )
+    MESSAGE(FATAL_ERROR
+      "Could not find libprotobuf-c package. Check build instructions: https://github.com/nats-io/nats.c#building-with-streaming")
   ENDIF()
   add_definitions(-DNATS_HAS_STREAMING)
   set(NATS_DOC_INCLUDE_STREAMING "NATS_HAS_STREAMING")
@@ -121,6 +123,8 @@ if(NATS_BUILD_USE_SODIUM)
     SET(NATS_SODIUM_DIR "" CACHE STRING
       "An optional hint to a directory for finding `libsodium`"
     )
+    MESSAGE(FATAL_ERROR
+      "Could not find libsodium package. Check build instructions: https://github.com/nats-io/nats.c#building-with-libsodium")
   ENDIF()
   add_definitions(-DNATS_USE_LIBSODIUM)
 endif(NATS_BUILD_USE_SODIUM)

